### DISCUSSION TITLE
How can I use 1  Tabpy Python Script to get different dataframe based on _args in Tableau prep

### DISCUSSION THREAD

**Discussion by Saketkr21** [tableau/TabPy#636] [Q&A]
**Problem Statement : Currently, Tableau Prep Script does not support passing args, also no args can be passed to  _get_output_schema_ function** 
I have 1 generic python function inside script that sends different kinds of pandas dataframe based on input args. 
`get_collection_data(mongodb collection name, db_name)`

I figured out a way to create dynamic number of functions based on how many collections are in mongoDB.
`Looping over all collections available and coll<collectionName> = lambda x : get_collection_data()`

But, I could not get a work around with _get_output_schema_ function. I am able to generate schema for different collections  pd.DataFrame({"dynamic schema based on collection name"}), But could not change _get_output_schema_ function behaviour dynamically on Run-Time.
I tried with exec(), eval().. But No luck.

This could be solved if I can pass _args to the both function call (get_collection_data and get_output_schema)

Right now due to this limitation, I have to manage multiple .py files with almost 98% same code. Only the _get_output_schema_ function is different.

Can anyone please suggest a better way to do this ?
Note : I tried to use MongoDB BI Connector, It has a lot more complexity. I find this way much easier.

<img width="210" alt="image" src="https://github.com/tableau/TabPy/assets/35528409/94dadbd2-d52a-4ec5-8586-de7bc054b20d">

**Comment by nmannheimer** [comment]
Unfortunately, this is going to be a limitation on the Tableau Prep side rather than TabPy, since Prep is what requires knowledge of the Schema.

I'd suggest submitting this as a feature request for Prep and potentially exploring using Table Extensions in Desktop/Web Authoring as a way to access the data dynamically using Python. This will likely be a major change in Prep though, so the advice they will likely give is to work on using the MongoDB connector.

**Reply by Unwrittend** [reply]
Does this issue still exist? I'm trying to make a script to dynamically remove various fields from a dataset based on arguments, but my schema can't be pre-defined. It is defined after fields have been removed. Is there really no way to do this?
